junit-platform-surefire-provider/src/main/java/org/junit/platform/surefire/provider/JUnitPlatformProvider.java

```diff
@@ -134,15 +134,11 @@ private void invokeSingleClass(Class<?> testClass, RunListener runListener) {
 
 		Optional<List<String>> includes = getGroupsOrTags(getPropertiesList(INCLUDE_GROUPS),
 			getPropertiesList(INCLUDE_TAGS));
-		if (includes.isPresent()) {
-			filters.add(TagFilter.includeTags(includes.get()));
-		}
+		includes.ifPresent(includeFilters -> filters.add(TagFilter.includeTags(includeFilters)));
```

Comment:
When I originally called this variable `includeFilters` it was because the engine filters would be added at the same point.  Note that we decided that the code to include or exclude tags should be committed without waiting for the engine filter implementation.